source/common/umf_pools/disjoint_pool_config_parser.cpp

```diff
@@ -1,18 +1,27 @@
 //===--- disjoint_pool_config_parser.cpp -configuration for USM memory pool-==//
 //
+// Copyright (C) 2023-2024 Intel Corporation
+//
 // Part of the Unified-Runtime Project, under the Apache License v2.0 with LLVM Exceptions.
 // See https://llvm.org/LICENSE.txt for license information.
 // SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 //
 //===----------------------------------------------------------------------===//
 
 #include "disjoint_pool_config_parser.hpp"
+#include "logger/ur_logger.hpp"
 
 #include <iomanip>
 #include <iostream>
 #include <limits>
 #include <string>
 
+#define CheckConfigRet(umf_ret)                  \
+    if (umf_ret != UMF_RESULT_SUCCESS) {                                    \
+        logger::error("DisjointPool params failed");                        \
+        return;                                                             \
```

Comment:
Should we just ignore the error? If any of the setParams function fail and we will just exit the function we might use incomplete config later on.

For calls in DisjointPoolAllConfigs::DisjointPoolAllConfigs I think we should have an assert and for calls in parseDisjointPoolConfig if there is any error we should catch it, log it, and probably revert all configs to the default settings.

Reply:
Done